### PR TITLE
49976_Static_Pages

### PR DESCRIPTION
**Pull request by noor-framgia**

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,14 +1,26 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <title>Brs19</title>
+  <title><%= t(:home_head)%></title>
```
Comment (manazirahsan) [inline app/views/layouts/application.html.erb]: convention

File: app/views/user/user/_header.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to t(:home_head), root_path , id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to t(:my_profile),'#', class:"btn btn-default" %></li>
```
Comment (manazirahsan) [inline app/views/user/user/_header.html.erb]: double quote.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -2,4 +2,13 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  before_action :set_locale
+
+  def set_locale
+    I18n.locale = params[:locale] || I18n.default_locale
+  end
+
+  def default_url_options(options = {})
```
Comment (abdullahallmehedi) [inline app/controllers/application_controller.rb]: ()

File: app/views/admin/admin/_header.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+	<%= link_to t(:home_head), root_path , id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to t(:my_profile),'#', class:"btn btn-default" %></li>
```
Comment (abdullahallmehedi) [inline app/views/admin/admin/_header.html.erb]: ""

File: app/views/admin/admin/_header.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        <li><%= link_to t(:books),'#', class:"btn btn-default" %></li>
+        <li><%= link_to t(:users),'#', class:"btn btn-default" %></li>
+        <li><%= link_to t(:requests),'#', class:"btn btn-default" %></li>
+        <li><%= link_to t(:logout),'#', class:"btn btn-default" %></li>
```
Comment (abdullahallmehedi) [inline app/views/admin/admin/_header.html.erb]: ""

File: app/views/admin/static_pages/home.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<div	class="center	jumbotron">
```
Comment (abdullahallmehedi) [inline app/views/admin/static_pages/home.html.erb]: spacing

File: app/views/admin/static_pages/home.html.erb [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+  <div><p><strong><%= t(:admin)%></strong></p></div>
```
Comment (abdullahallmehedi) [inline app/views/admin/static_pages/home.html.erb]: ()

File: app/views/application/_footer.html.erb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<footer class="footer">
+  <div row>
+    <div class="col-md-4">
+      <p>&copy; <%= t :framgia %></p>
+    </div>
+  </div>
+      
+  <nav>
+    <ul>
+      <li><%= link_to t(:about),about_path %></li>
```
Comment (abdullahallmehedi) [inline app/views/application/_footer.html.erb]: spacing

File: app/views/application/_footer.html.erb [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+      <li><%= link_to t(:contact),contact_path %></li>
```
Comment (abdullahallmehedi) [inline app/views/application/_footer.html.erb]: spacing

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -1,14 +1,26 @@
...
   <%= csrf_meta_tags %>
+  <%= render "shim" %>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
 </head>
 <body>
-
-<%= yield %>
-
+<%= render "header" %>
+<div class="container">
```
Comment (SSalekin) [inline app/views/layouts/application.html.erb]: 2 space indentation.

File: app/views/layouts/application.html.erb [hunk 11]
```diff
@@ -1,14 +1,26 @@
...
+    <%= t :language %>:
+    <%= link_to_unless_current t(:english), locale: "en" %>
+  </div>
+  <% flash.each do |message_type, message| %>
+    <div class="alert alert-<%= message_type %>"><%= message %></div>
+  <% end %>
+  <%= yield %>
+  <%= render "footer" %>
+  <%= debug(params) if Rails.env.development? %>
+</div>
```
Comment (SSalekin) [inline app/views/layouts/application.html.erb]: indentation

File: app/views/static_pages/about.html.erb [hunk 12]
```diff
@@ -0,0 +1,5 @@
+<% provide :title, t(:about)%>
+<h1>About</h1>
```
Comment (SSalekin) [inline app/views/static_pages/about.html.erb]: i18n

File: app/views/static_pages/contact.html.erb [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<% provide :title, t(:about)%>
+<h1>Contact</h1>
```
Comment (SSalekin) [inline app/views/static_pages/contact.html.erb]: i18n

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,14 +1,26 @@
...
+  <div id="nav">
+    <%= t :language %>:
+    <%= link_to_unless_current t(:english), locale: "en" %>
+  </div>
+  <% flash.each do |message_type, message| %>
+    <div class="alert alert-<%= message_type %>"><%= message %></div>
+  <% end %>
+  <%= yield %>
+  <%= render "footer" %>
+  <%= debug(params) if Rails.env.development? %>
```
Comment (abdullahallmehedi) [inline app/views/layouts/application.html.erb]: we will use pry. remove this pls for now.

File: app/views/shared/_error_messages.html.erb [hunk 15]
```diff
@@ -0,0 +1,12 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      <%= t :form_contains %><%= pluralize(object.errors.count, t(:error)) %>.
```
Comment (abdullahallmehedi) [inline app/views/shared/_error_messages.html.erb]: ()

File: app/views/layouts/application.html.erb [hunk 16]
```diff
@@ -1,14 +1,26 @@
...
+<div class="container">
+  <div id="nav">
+    <%= t :language %>:
+    <%= link_to_unless_current t(:english), locale: "en" %>
+  </div>
+  <% flash.each do |message_type, message| %>
+    <div class="alert alert-<%= message_type %>"><%= message %></div>
+  <% end %>
+  <%= yield %>
+  <%= render "footer" %>
```
Comment (abdullahallmehedi) [inline app/views/layouts/application.html.erb]: try symbol

File: app/views/layouts/application.html.erb [hunk 17]
```diff
@@ -1,14 +1,26 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <title>Brs19</title>
+  <title><%= t(:home_head)%></title>
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
```
Comment (SSalekin) [inline app/views/layouts/application.html.erb]: use double quote

File: app/views/static_pages/about.html.erb [hunk 18]
```diff
@@ -0,0 +1,5 @@
...
+<p>
+  <a href="http://www.framgia.com"><%= t(:home_head)%></a>
```
Comment (abdullahallmehedi) [inline app/views/static_pages/about.html.erb]: ()

File: app/views/static_pages/home.html.erb [hunk 19]
```diff
@@ -0,0 +1,6 @@
+<div	class="center	jumbotron">
```
Comment (SSalekin) [inline app/views/static_pages/home.html.erb]: spacing

File: app/views/static_pages/contact.html.erb [hunk 20]
```diff
@@ -0,0 +1,10 @@
+<% provide :title, t(:about)%>
```
Comment (abdullahallmehedi) [inline app/views/static_pages/contact.html.erb]: spacing

File: app/views/static_pages/contact.html.erb [hunk 21]
```diff
@@ -0,0 +1,10 @@
...
+<h1>Contact</h1>
+<p>
+  <h2><%= t(:home_head)%> By</h2>
```
Comment (abdullahallmehedi) [inline app/views/static_pages/contact.html.erb]: ()

File: app/views/static_pages/home.html.erb [hunk 22]
```diff
@@ -0,0 +1,6 @@
...
+  <h1> <%= t :home_head %></h1>
```
Comment (SSalekin) [inline app/views/static_pages/home.html.erb]: <h1><
won't have space

File: app/views/static_pages/home.html.erb (hunk 19 above)
Comment (abdullahallmehedi) [inline app/views/static_pages/home.html.erb]: spacing

File: app/views/user/static_pages/home.html.erb [hunk 23]
```diff
@@ -0,0 +1,7 @@
+<div	class="center	jumbotron">
```
Comment (abdullahallmehedi) [inline app/views/user/static_pages/home.html.erb]: spacing

File: app/views/user/static_pages/home.html.erb [hunk 24]
```diff
@@ -0,0 +1,7 @@
...
+  <div><p><strong><%= t(:user)%></strong></p></div>
```
Comment (abdullahallmehedi) [inline app/views/user/static_pages/home.html.erb]: () + spacing

File: app/views/user/static_pages/home.html.erb (hunk 23 above)
Comment (SSalekin) [inline app/views/user/static_pages/home.html.erb]: spacing.

File: config/routes.rb [hunk 25]
```diff
@@ -1,4 +1,18 @@
 Rails.application.routes.draw do
+  scope "(:locale)", :locale => /en/ do
+    root "static_pages#home"
+    get "home"=> "static_pages#home"
+    get "about"=> "static_pages#about"
+    get "contact"=> "static_pages#contact"
+
+    namespace :admin do
+      get "home"=> "static_pages#home"
```
Comment (abdullahallmehedi) [inline config/routes.rb]: root

File: app/views/user/static_pages/home.html.erb [hunk 26]
```diff
@@ -0,0 +1,7 @@
...
+  <h1> <%= t :home_head %></h1>
```
Comment (SSalekin) [inline app/views/user/static_pages/home.html.erb]: no space between h1 and <%=

File: app/views/static_pages/about.html.erb (hunk 12 above)
Comment (SSalekin) [inline app/views/static_pages/about.html.erb]: i18n

File: app/views/layouts/application.html.erb [hunk 27]
```diff
@@ -1,14 +1,26 @@
...
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
```
Comment (SSalekin) [inline app/views/layouts/application.html.erb]: double quote

File: app/views/application/_footer.html.erb [hunk 28]
```diff
@@ -0,0 +1,13 @@
+<footer class="footer">
+  <div row>
+    <div class="col-md-4">
+      <p>&copy; <%= t :framgia %></p>
+    </div>
+  </div>
+  <nav>
+    <ul>
+      <li><%= link_to t(:about),about_path %></li>
```
Comment (abdullahallmehedi) [inline app/views/application/_footer.html.erb]: spacing

File: app/views/application/_footer.html.erb [hunk 29]
```diff
@@ -0,0 +1,13 @@
...
+      <li><%= link_to t(:contact),contact_path %></li>
```
Comment (abdullahallmehedi) [inline app/views/application/_footer.html.erb]: spacing

File: app/views/layouts/application.html.erb [hunk 30]
```diff
@@ -1,14 +1,25 @@
...
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
-  <%= csrf_meta_tags %>
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= t(:home_head)%></title>
```
Comment (abdullahallmehedi) [inline app/views/layouts/application.html.erb]: ()

File: app/views/layouts/application.html.erb [hunk 31]
```diff
@@ -1,14 +1,25 @@
...
+    <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
```
Comment (abdullahallmehedi) [inline app/views/layouts/application.html.erb]: ""

File: app/views/layouts/application.html.erb [hunk 32]
```diff
@@ -1,14 +1,25 @@
...
+    <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
```
Comment (abdullahallmehedi) [inline app/views/layouts/application.html.erb]: ""

File: app/views/layouts/application.html.erb (hunk 31 above)
Comment (manazirahsan) [inline app/views/layouts/application.html.erb]: double qoute.

File: app/views/application/_header.html.erb [hunk 33]
```diff
@@ -0,0 +1,10 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to t(:home_head), root_path , id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <%= link_to t(:sign_in), "#", class:"btn btn-default" %>
```
Comment (abdullahallmehedi) [inline app/views/application/_header.html.erb]: spacing

File: config/routes.rb [hunk 34]
```diff
@@ -1,4 +1,17 @@
 Rails.application.routes.draw do
+  scope "(:locale)", :locale => /en/ do
```
Comment (SSalekin) [inline config/routes.rb]: convention.

**Comment by SSalekin** [comment]
ok

**Comment by abdullahallmehedi** [comment]
ok

**Comment by manazirahsan** [comment]
okey

**Comment by SSalekin** [comment]
ok

**Pull request merged**